geopandas/geodataframe.py

```diff
@@ -586,8 +589,27 @@ def from_features(cls, features, crs=None, columns=None):
             }
             # load properties
             row.update(feature["properties"])
+            if "id" in feature:
+                row.update({"id": feature["id"]})
             rows.append(row)
-        return GeoDataFrame(rows, columns=columns, crs=crs)
+
+        gdf = GeoDataFrame(rows, columns=columns, crs=crs)
+        if id_as_index:
+            if "id" not in gdf:
+                warnings.warn(
+                    "Argument `id_as_index `is set to True but there is"
+                    " no `id` field in the given features."
```

Comment:
Can you expand this and the other warning with a bit more explicit explanation of what geopandas is doing? something like `id_as_index=True is ignored`.